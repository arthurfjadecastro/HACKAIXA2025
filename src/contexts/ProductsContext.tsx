import React, { createContext, useContext, useCallback, useState } from 'react';

interface ProductsContextType {
  refreshTrigger: number;
  triggerRefresh: () => void;
}

const ProductsContext = createContext<ProductsContextType | undefined>(undefined);

interface ProductsProviderProps {
  children: React.ReactNode;
}

export const ProductsProvider: React.FC<ProductsProviderProps> = ({ children }) => {
  const [refreshTrigger, setRefreshTrigger] = useState(0);

  const triggerRefresh = useCallback(() => {
    setRefreshTrigger(prev => prev + 1);
  }, []);

  return (
    <ProductsContext.Provider
      value={{
        refreshTrigger,
        triggerRefresh,
      }}
    >
      {children}
    </ProductsContext.Provider>
  );
};

export const useProductsContext = (): ProductsContextType => {
  const context = useContext(ProductsContext);
  if (!context) {
    throw new Error('useProductsContext deve ser usado dentro de ProductsProvider');
  }
  return context;
};
