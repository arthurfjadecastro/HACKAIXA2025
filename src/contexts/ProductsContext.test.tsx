import React from 'react';
import { render } from '@testing-library/react-native';
import { ProductsProvider } from './ProductsContext';

// Mock das dependÃªncias
jest.mock('../services', () => ({
  productsApi: {
    getProducts: jest.fn(),
    createProduct: jest.fn(),
    updateProduct: jest.fn(),
    deleteProduct: jest.fn(),
  },
}));

const TestComponent = () => <div data-testid="test-component">Test</div>;

describe('ProductsContext', () => {
  it('should export ProductsProvider', () => {
    expect(ProductsProvider).toBeDefined();
    expect(typeof ProductsProvider).toBe('function');
  });

  it('should provide context to children', () => {
    const { toJSON } = render(
      <ProductsProvider>
        <TestComponent />
      </ProductsProvider>
    );
    expect(toJSON()).toBeDefined();
  });

  it('should be a valid React component', () => {
    expect(React.isValidElement(<ProductsProvider><TestComponent /></ProductsProvider>)).toBe(true);
  });

  it('should handle children prop', () => {
    const { toJSON } = render(
      <ProductsProvider>
        <div>Child 1</div>
        <div>Child 2</div>
      </ProductsProvider>
    );
    expect(toJSON()).toBeDefined();
  });

  it('should match snapshot', () => {
    const { toJSON } = render(
      <ProductsProvider>
        <TestComponent />
      </ProductsProvider>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
