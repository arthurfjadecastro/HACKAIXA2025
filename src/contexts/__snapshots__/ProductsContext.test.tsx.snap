// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsContext should match snapshot 1`] = `
<div
  data-testid="test-component"
>
  Test
</div>
`;
